index.js

```diff
@@ -1,46 +1,102 @@
 import process from 'node:process';
 
 const callbacks = new Set();
+const callbacksSync = new Set();
+
 let isCalled = false;
 let isRegistered = false;
 
-function exit(shouldManuallyExit, signal) {
+function exit(shouldManuallyExit, isSynchronous, signal) {
+	if (callbacks.size > 0 && isSynchronous) {
+		console.error('SYNCHRONOUS TERMINATION NOTICE:');
+		console.error('When explicitly exiting the process via process.exit, asynchronous');
+		console.error('tasks in your exitHooks will not run. Either remove these tasks,');
+		console.error('or use exitHook.exit() instead.');
```

Comment:
Use a single `console.error` call.

Reply:
You are assuming that this package will only be used at the top-level. This is gonna create a lot of problems if this package is used in dependencies you don't control, and then `process.exit()` is used in top-level code.